examples/counter-react/src/counterByName.js

```diff
@@ -0,0 +1,26 @@
+import React, { useContext } from 'react';
+import { CounterStore } from './store';
+
+const Counter = ({ name }) => {
+  const [counterContext] = useContext(CounterStore.Context);
+  let { count } = counterContext.state;
+
+  // this isn't necessary, but this demonstrates how a store ref would be used if a storeKey is provided
+  const incrementCount = () => CounterStore[name].incrementCount();
+
+  return (
+    <div data-testid="counter-wrap">
+      <p>
+        Counter with key {name}: {count}
+      </p>
+      <button data-testid="decrement" onClick={counterContext.decrementCount}>
+        -
+      </button>
+      <button data-testid="increment" onClick={incrementCount}>
+        +
+      </button>
+    </div>
+  );
+};
+
+export default CounterStore.Provider(Counter);
```

Comment:
Each Counter gets their own provider, but providing the storeKey to the wrapped component means that the CounterStore ref can be used to access the state or functions of any nested store, like `CounterStore['A'].incrementCount()`.

Reply:
🚀